### PR TITLE
r-functional: Add the functional R package.

### PR DESCRIPTION
**Pull request by jgaeb**

### CODE REVIEW

File: var/spack/repos/builtin/packages/r-functional/package.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class RFunctional(RPackage):
+    """Curry, Compose, and other higher-order functions
+
+    Curry, Compose, and other higher-order functions"""
+
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/r-functional/package.py:13]: Mind adding the `homepage` property?

**Review by tldahlgren — APPROVED** [review]
Confirmed all version sha256 and am able to successfully build the package.  Would like to see the homepage added though.

File: var/spack/repos/builtin/packages/r-functional/package.py (hunk 1 above)
Comment (jgaeb) [reply var/spack/repos/builtin/packages/r-functional/package.py:13]: Somewhat unusually, it doesn't have a homepage or associated GitHub repo. Should I just set the homepage to its [CRAN homepage](https://cran.r-project.org/web/packages/functional/index.html)? I'm not sure what the convention is for packages like that.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/r-functional/package.py:13]: Ick.  It does provide a link to the reference manual but not sure it's worth adding so I'll leave that up to you.

**Comment by tldahlgren** [comment]
Updated branch to pick up CI fix.

**Pull request merged**